cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -131,43 +131,39 @@ __device__ __forceinline__ uint8_t decode_digit(char c, bool* valid_flag)
 /**
  * @brief Parses a character string and returns its numeric value.
  *
- * @param data The character string for parse
- * @param start The index within data to start parsing from
- * @param end The end index within data to end parsing
- * @param opts The global parsing behavior options
- * @param base Base (radix) to use for conversion
+ * @param[in] begin Beginning of the character string
+ * @param[in] end End of the character string
+ * @param[in] opts The global parsing behavior options
+ * @tparam base Base (radix) to use for conversion
  *
  * @return The parsed and converted value
  */
 template <typename T, int base = 10>
-__inline__ __device__ T
-parse_numeric(const char* data, long start, long end, ParseOptions const& opts)
+__inline__ __device__ T parse_numeric(const char* begin, const char* end, ParseOptions const& opts)
 {
   T value{};
   bool all_digits_valid = true;
 
   // Handle negative values if necessary
   int32_t sign = 1;
-  if (data[start] == '-') {
+  if (*begin == '-') {
     sign = -1;
-    start++;
+    begin++;
   }
 
   // Skip over the "0x" prefix for hex notation
-  if (base == 16 && start + 2 <= end && data[start] == '0' && data[start + 1] == 'x') {
-    start += 2;
-  }
+  if (base == 16 && begin + 2 <= end && *begin == '0' && *(begin + 1) == 'x') { begin += 2; }
 
   // Handle the whole part of the number
-  long index = start;
+  auto index = begin;
```

Comment:
let's not call this `index` anymore. What about `iter` or `ptr`.

Reply:
Yeah, good point. We can actually remove it and just use begin.